File: herencia-dto/SergioDeIscar-herencia-java/SergioDeIscar-herencia-java/src/main/java/org/example/repositories/CrudRepository.java
package org.example.repositories;

import java.util.List;

public interface CrudRepository<T, ID> {
    List<T> getAll();
    T getById(ID id);
    T save(T element, boolean storage);
    void saveAll(List<T> elements, boolean storage);
    T deleteById(ID id);
    T update(T element);
    T updateById(ID id, T element);
}

File: herencia-dto/SergioDeIscar-herencia-kotlin/SergioDeIscar-herencia-kotlin/src/main/kotlin/repositories/CrudRepository.kt
package repositories

interface CrudRepository<T,ID> {
    fun findAll(): Iterable<T>
    fun findById(id: ID): T?
    fun save(element: T): T
    fun saveAll(elements: Iterable<T>)
    fun deleteById(id: ID): Boolean
    fun delete(element: T): Boolean
    fun deleteAll()
    fun existsById(id: ID): Boolean
}